python/ray/serve/_private/deployment_state.py

```diff
@@ -2717,13 +2724,10 @@ def update(self) -> bool:
             and len(ds._replicas.get()) == ds.target_num_replicas
             for ds in self._deployment_states.values()
         )
-        draining_nodes = self._cluster_node_info_cache.get_draining_nodes()
-        if (
-            not draining_nodes
-            and RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY
-            and allow_active_compaction
-        ):
-            compact_opp = self._deployment_scheduler.detect_compact_opportunities()
+        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY:
+            compact_opp = self._deployment_scheduler.detect_compact_opportunities(
+                allow_active_compaction=allow_active_compaction
+            )
             if compact_opp:
```

Comment:
I have no idea what `compact_opp` means here 😓

Reply:
would need to go read the other code to figure out what this thing actually is

Reply:
so at a minimum let's add a type hint, but I also feel the naming can be made more obvious (see suggestion above)

Reply:
Renamed to `node_info` and added type hint (type hint is just Tuple right now). Also added a comment, hopefully it makes it clearer.